From 0af3f8871ef00eea6f402fc6588bb225b9c8ffbc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo@beauzee.fr>
Date: Mon, 26 Mar 2018 16:44:44 +0200
Subject: [PATCH 07/17] modules:common: Use the full module name as MODULE_NAME

This avoid conflicts when linking modules staticly on platforms that don't have objdump
---
 modules/common.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/common.am b/modules/common.am
index b991f6ce2f..d327b23871 100644
--- a/modules/common.am
+++ b/modules/common.am
@@ -17,12 +17,13 @@ LTLIBVLCCORE = $(top_builddir)/src/libvlccore.la
 # Module name from object or executable file name.
 MODULE_STRING = $$(p="$@"; p="$${p\#\#*/}"; p="$${p\#lib}"; p="$${p%_plugin*}"; p=$$(echo "$$p"|sed 's/-/_/g'); p="$${p%.lo}"; echo "$$p")
 MODULE_NAME = $$(p="$@"; p=$$(echo "$$p"|sed 's/lib//g'); p="$${p%_plugin*}"; p=$$(echo "$$p"|sed 's/[-\/]/_/g'); p="$${p%.lo}"; echo "$$p")
+MODULE_FULLNAME = $$(p="$@"; p="$${p\#lib}"; p="$${p%_plugin*}"; p=$$(echo "$$p"|sed 's/[-\/]/_/g'); p="$${p%.lo}"; echo "$$p")
 
 AM_CPPFLAGS = -DMODULE_STRING=\"$(MODULE_STRING)\"
 if HAVE_DYNAMIC_PLUGINS
 AM_CPPFLAGS += -D__PLUGIN__
 else
-AM_CPPFLAGS += -DMODULE_NAME=$(MODULE_NAME)
+AM_CPPFLAGS += -DMODULE_NAME=$(MODULE_FULLNAME)
 endif
 AM_CFLAGS =
 AM_CXXFLAGS =
-- 
2.21.1 (Apple Git-122.3)

